qa/vagrant/src/test/resources/packaging/scripts/plugin_test_cases.bash

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bats
+
+# This file is used to test the installation and removal
+# of plugins with a tar gz archive.
+
+# WARNING: This testing file must be executed as root and can
+# dramatically change your system. It removes the 'elasticsearch'
+# user/group and also many directories. Do not execute this file
+# unless you know exactly what you are doing.
+
+# The test case can be executed with the Bash Automated
+# Testing System tool available at https://github.com/sstephenson/bats
+# Thanks to Sam Stephenson!
+
+# Licensed to Elasticsearch under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+##################################
+# Common test cases for both tar and rpm/deb based plugin tests
+##################################
+# This file is symlinked to both 25_tar_plugins.bats and 50_plugins.bats so its
+# executed twice - once to test plugins using the tar distribution and once to
+# test files using the rpm distribution or the deb distribution, whichever the
+# system uses.
+
+# Load test utilities
+load packaging_test_utils
+load plugins
+
+setup() {
+    # The rules on when we should clean an reinstall are complex - all the
+    # jvm-example tests need cleaning because they are rough on the filesystem.
+    # The first and any tests that find themselves without an ESHOME need to
+    # clean as well.... this is going to mostly only happen on the first
+    # non-jvm-example-plugin-test _and_ any first test if you comment out the
+    # other tests. Commenting out lots of test cases seems like a reasonably
+    # common workflow.
+    if [ $BATS_TEST_NUMBER == 1 ] ||
+            [[ $BATS_TEST_NAME =~ install_jvm.*example ]] ||
+            [ ! -d "$ESHOME" ]; then
+        echo "cleaning" >> /tmp/ss
+        clean_before_test
+        install
+    fi
+}
+
+if [[ "$BATS_TEST_FILENAME" =~ 25_tar_plugins.bats$ ]]; then
+    load tar
+    GROUP='TAR PLUGINS'
+    install() {
+        install_archive
+        verify_archive_installation
+    }
+    export ESHOME=/tmp/elasticsearch
+    export_elasticsearch_paths
+else
+    if is_rpm; then
+        GROUP='RPM PLUGINS'
+    elif is_dpkg; then
+        GROUP='DEB PLUGINS'
+    fi
+    export ESHOME="/usr/share/elasticsearch"
```

Comment:
That would have been great to use ${packaging.elasticsearch.home.dir} var here but I guess it's not possible